src/main/scala/strawman/collection/Map.scala

```diff
@@ -0,0 +1,44 @@
+package strawman.collection
+
+import strawman.collection.mutable.Builder
+
+import scala.Option
+import scala.annotation.unchecked.uncheckedVariance
+import scala.Predef.???
+
+/** Base Map type */
+trait Map[K, +V]
+  extends Iterable[(K, V)]
```

Comment:
@sjrd You're right. You would need to use a Product2 extractor instead of the language Tuple2 extractor. Another example of the (..) tuple syntax swaying one's hand, yours in this case.

@odersky I think what is "least power" depends on whose side you're looking, the producer or the consumer of the type. Lots of times you're the only implementor, so you only have consumers. But in this case it's a collections library trait, so you have both. So it's a design decision/judgement call.